cmd/thanos/rule.go

```diff
@@ -963,3 +1035,88 @@ func validateTemplate(tmplStr string) error {
 	}
 	return nil
 }
+
+func newEndpointSet(
```

Comment:
Maybe this can be called `prepareEndpointSet` or similar to avoid having a similar name to `query.NewEndpointSet`?

Reply:
Done